### DISCUSSION TITLE
`airflow dags trigger -e {date}` producing incorrect logical_date.start / logical_date.end values

### DISCUSSION THREAD

**Discussion by harrisjoseph** [apache/airflow#25687] [Q&A] [answered]
This issue may relate to https://github.com/apache/airflow/discussions/21577

When running a command like `airflow dags trigger -e 2022-02-14 {dag_id}`, on a dag with a weekly cron schedule `0 0 * * 1`, I'm finding that the data interval is not as expected. In this instance I get:
```
Data Interval:
Start: 2022-01-31, 00:00:00 UTC
End: 2022-02-07, 00:00:00 UTC
```
where I would expect `2022-02-14 / 2022-02-21`

This is causing issues with the underlying runs referring to data in the incorrect period.

I know that ultimately this is calculated by `infer_manual_data_interval`, but I'm not sure where this function is going wrong in determining the correct interval from the range. I've tested the code for this locally, and found the results are strange:

```
from airflow.timetables.interval import CronDataIntervalTimetable
import pytz

dt = datetime(2022,2,14, tzinfo=pytz.UTC)
tt = CronDataIntervalTimetable('0 0 * * 1', pytz.UTC)
tt.infer_manual_data_interval(run_after=dt)

DataInterval(start=DateTime(2022, 1, 31, 0, 0, 0, tzinfo=Timezone('UTC')), end=DateTime(2022, 2, 7, 0, 0, 0, tzinfo=Timezone('UTC')))
```

I know that the execution_date has been deprecated in favour of data_interval_start / data_interval_end, and the `DagRun` object expects a 2-tuple data_interval argument, but the API endpoint still only expects a singular execution_date parameter, so I'm unable to set the correct data interval manually without writing my own custom dag.create_dagrun(...) codepath for this.

---

Two thoughts here:
1. Is infer_manual_data_interval behaving as expected? Do we need a new test case + a bug fix?
2. Should we just move towards explicitly specifying data_interval_start / data_interval_end in the API 

Any input from someone familiar with the new timetable code + the changes around data_interval / execution_date / logical_date would be much appreciated.

Many thanks,
Joe

**Comment by potiuk** [comment, marked as answer]
I think it needs a new "airflow dags trigger" and new API simply. Execution date is indeed deprecated. I think creating an issue for that is a good idea, so feel free.

**Comment by harrisjoseph** [comment]
Underlying bug is fixed in: https://github.com/apache/airflow/issues/23689, which should be released in the next version
